component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/SchemaImpl.java

```diff
@@ -67,6 +65,48 @@
         entriesOrder = EntriesOrder.of(getFieldsOrder());
     }
 
+    /**
+     * Optimized hashcode method (do not enter inside field hashcode, just getName, ignore props fields).
+     * 
+     * @return hashcode.
+     */
+    @Override
+    public int hashCode() {
+        final String e1 =
+                this.entries != null ? this.entries.stream().map(Entry::getName).collect(Collectors.joining(",")) : "";
+        final String m1 = this.metadataEntries != null
+                ? this.metadataEntries.stream().map(Entry::getName).collect(Collectors.joining(","))
+                : "";
+
+        return Objects.hash(this.type,
+                this.elementSchema,
+                e1,
+                m1);
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        if (!(obj instanceof SchemaImpl)) {
+            return false;
+        }
+        final SchemaImpl other = (SchemaImpl) obj;
+        if (!other.canEqual(this)) {
+            return false;
+        }
+        return Objects.equals(this.type, other.type)
+                && Objects.equals(this.elementSchema, other.elementSchema)
+                && Objects.equals(this.entries, other.entries)
+                && Objects.equals(this.metadataEntries, other.metadataEntries)
+                && Objects.equals(this.props, other.props);
+    }
+
+    protected boolean canEqual(final SchemaImpl other) {
+        return true;
```

Comment:
Don't get the purpose w/ upper usage...

Reply:
It's special for good equals method, see https://www.artima.com/articles/how-to-write-an-equality-method-in-java for example.
Lombok generate a 
```java
protected boolean canEqual(final Object other) {
    return other instanceof SchemaImpl;
}
```
and used by equals as : 
```
if (other instanceof SchemaImpl) {
   if (! (SchemaImpl.canEqual(this)) { return false; }
   ...
}
```
(but as it's used with this that is at least SchemaImpl, i rewrite)
On SchemaImpl subclass, you rewrite as : 
```java
protected boolean canEqual(final SchemaImpl other) {
    return other instanceof SchemaImplSubClass;
}
```

Reply:
ok got it.